catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +171,62 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService("dog_legs.sch", "dog_paws.sch");
+        // test warnings
+        try {
+            service.validate(getMetacard("dog_4leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            for (String warning : ex.getWarnings()) {
+                assertThat(warning.equals(SchematronValidationService.sanitize(warning)), is(true));
+            }
+        }
+        // test errors
+        try {
+            service.validate(getMetacard("dog_3leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            for (String error : ex.getErrors()) {
+                assertThat(error.equals(SchematronValidationService.sanitize(error)), is(true));
+            }
+        }
+        // sanitize is called twice for each validate call
+        verify(service, times(4)).sanitize(anyString());
+    }
+
+    @Test
+    public void testSanitization() throws SchematronInitializationException {
+        // verify that nothing changes
```

Comment:
I thought about this test a little bit more. I think it would make sense to split it into two tests, a test that asserts nothing changes (you could use the assertion on line 203 and 206), a test that asserts things change (the rest of the tests)

That being said, I don't think you need the assertions on 216 through the end, i think up through 214 is enough